tensorflow/python/keras/optimizers.py

```diff
@@ -89,7 +89,7 @@ def get_gradients(self, loss, params):
           function not implemented).
     """
     grads = K.gradients(loss, params)
-    if None in grads:
+    if any([g is None for g in grads]):
```

Comment:
@OverLordGoldDragon 
This is very interesting. Thank you for the fix. 
Though I cannot really see how these two lines lead to different outcomes. 
So an iterable may contain a `None` while none of its members is a `None`? 
Can you shed a light or point me in the right direction to understand what's going or to recreate a simple example? 
Thanks in advance.

Reply:
@engheta `None in grads` is same as `any([g == None for g in grads])`; problem is, `g` may be a tf.Tensor/tf.Variable which doesn't have [`.__eq__`](https://docs.python.org/3/reference/datamodel.html#object.__eq__) defined, so `is None` must be used instead.

Reply:
So the lack of a `.__eq__` method could cause `g == None` to resolve to `True` even though `g is None` is in fact `False`?

Reply:
@engheta [Updated](https://stackoverflow.com/a/58279661/10133797); to note, I'm not certain whether some TF objects actually don't have `.__eq__` defined at all, or are simply redefined.

Reply:
Yes, makes sense. After looking more closely, I realized that the error is caused by TF's attempt to convert the `None` to a tensor for the comparison. 

I like (and upvoted) the Stack Overflow entry and your simple and effective fix here. 

The banes and pleasures of using open-source software...